src/VisualStudio/Core/Def/DocumentOutline/DocumentOutlineViewModel.cs

```diff
@@ -0,0 +1,313 @@
+                }
+
+                if (searchText is { } currentQuery)
+                {
+                    ImmutableArray<DocumentSymbolDataViewModel> documentSymbolViewModelItems;
+                    if (currentQuery == string.Empty)
+                    {
+                        // search was cleared, show all data.
+                        documentSymbolViewModelItems = DocumentOutlineHelper.GetDocumentSymbolItemViewModels(model.DocumentSymbolData);
+                        ApplyExpansionStateToNewItems(documentSymbolViewModelItems, DocumentSymbolViewModelItems);
+                    }
+                    else
+                    {
+                        // We are going to show results so we unset any expand / collapse state
+                        // If we are in the middle of searching the developer should always be able to see the results
+                        // so we don't want to collapse (and therefore hide) data here.
+                        documentSymbolViewModelItems = DocumentOutlineHelper.GetDocumentSymbolItemViewModels(
+                             DocumentOutlineHelper.SearchDocumentSymbolData(model.DocumentSymbolData, currentQuery, cancellationToken));
+                    }
+
+                    DocumentSymbolViewModelItems = documentSymbolViewModelItems;
+                }
+
+                if (position is { } currentPosition)
+                {
+                    var caretPoint = currentPosition.Point.GetPoint(_textBuffer, PositionAffinity.Predecessor);
+                    // If the we can't find the caret then the document has changed since we were queued to the point that we can't find this position.
+                    if (caretPoint.HasValue)
+                    {
+                        var symbolToSelect = DocumentOutlineHelper.GetDocumentNodeToSelect(DocumentSymbolViewModelItems, model.OriginalSnapshot, caretPoint.Value);
+                        // If null this means we can't find the symbol to select. The document has changed enough since we are queued that we can't find anything applicable.
+                        if (symbolToSelect is not null)
+                        {
+                            DocumentOutlineHelper.ExpandAncestors(DocumentSymbolViewModelItems, symbolToSelect.Data.SelectionRangeSpan);
+                            symbolToSelect.IsSelected = true;
+                        }
+                    }
+                }
+            }
+
+            // If we aren't filtering to search results do expand/collapse
+            if (expansion is { } expansionOption && string.IsNullOrEmpty(searchText))
+            {
+                DocumentOutlineHelper.SetExpansionOption(DocumentSymbolViewModelItems, expansionOption);
+            }
+            else if (expansion is { } shouldExpand)
+            {
+                DocumentOutlineHelper.SetExpansionOption(DocumentSymbolViewModelItems, shouldExpand);
+            }
```

Comment:
afaict the two blocks are identical.  can you explain the different here @jmarolf ?